Aligent/LiveChatBundle/Service/Webhook/ChatEnd.php

```diff
@@ -85,12 +88,11 @@ public function handleRequest($jsonString) {
      * Extract the required fields from the chatEnd request payload.
      *
      * @param $jsonString string Raw JSON from request
-     * @return ChatEndData
+     * @return array Parsed JSON data
      * @throws ChatException
      */
-    public function parseChatWebhook($jsonString) {
-        $jsonData = parent::parseChatWebhook($jsonString);
-        $chatEndData = new ChatEndData();
+    public function parseChatWebhook($jsonString, AbstractDTO $chatEndData) {
```

Comment:
Might be worth converting this to use Symfony's deserialize to object (https://symfony.com/doc/2.8/components/serializer.html#deserializing-an-object)